src/index.tsx

```diff
@@ -157,15 +148,68 @@ var saveToWebmarkFolder = (url: string | undefined, title: string | undefined):
     );
 }
 
-var getRandomUrlFromFolder = (): string => {
-    //TODO: implement getRandomUrlFromFolder
-    return 'https://example.com/random';
+let loadRandomUrlFromFolder = (): void => {
+    chrome.storage.sync.get(
+        ['webmarkFolderId'],
+        (result?) => {
+            if (Object.keys(result).length === 0) {
+                console.log('webmarkFolderId not found.');
+                return;
+            }
+            let webmarkFolderId: string = result!['webmarkFolderId'];
+            chrome.bookmarks.get(
+                webmarkFolderId,
+                () => {
+                    if (chrome.runtime.lastError) {
+                        console.log('webmark folder not found.');
+                        showNotice("Invalid Access");
+                        return;
+                    }
+                    let webmarkFolderId: string = result!['webmarkFolderId'];
+                    chrome.bookmarks.getSubTree(
+                        webmarkFolderId,
+                        (bookmarkTreeNodes: chrome.bookmarks.BookmarkTreeNode[]) => {
+                            let urlList: Array<string> = [];
+                            for (let node of bookmarkTreeNodes) {
+                                recursiveUrlCollection(node, urlList);
+                            }
+                            let randomIndex: number = Math.floor(Math.random() * urlList.length);
+                            let randomUrl: string = urlList[randomIndex];
+                            chrome.storage.sync.get(
```

Comment:
This part should definitely be a separate function, but this was left unrefactored because I didn't refactor it beforehand. You can skip this in the pull request.